File: API/Api.Domain/Product.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Api.Domain
{
    public class Product
    {
       

        public Guid product_id { get; set; }
        public string category_id { get; set; }
        public string product_name { get; set; }
        public string subcat_id { get; set; }
        public string size_id { get; set; }
        public string color_id { get; set; }
        public string product_img { get; set; }
        public Boolean is_main_page { get; set; }
        public Boolean is_new_product { get; set; }
        public Boolean is_best_selling { get; set; }
        public Boolean is_featured_product { get; set; }
        public Boolean is_on_sale { get; set; }
        public decimal price { get; set; }


    }
}

File: Frontend/ShoppersDelight/src/app/shared/classes/product.ts
// Products
export interface Product {
    id?: number;
    title?: string;
    description?: string;
    type?: string;
    brand?: string;
    collection?: any[];
    category?: string;
    price?: number;
    sale?: boolean;
    discount?: number;
    stock?: number;
    new?: boolean;
    quantity?: number;
    tags?: any[];
    variants?: Variants[];
    images?: Images[];
    image_src?: string;
}



export interface ProductList {
    product_id?: any;
    category_name?: string;
    product_name?: string;
    image_src?: string;
    subcat_name?: string;
    size_name?: string;
    product_img? : string;
    is_main_page?: boolean;
    is_new_product?: boolean;
    is_best_selling?: boolean;
    is_featured_product?: boolean;
    is_on_sale?: boolean;
    price?: number;
   description?: string;
   product_image?: string
   ProductDetail?: Image[];


}
export interface Image {
   
    image_src?: string;
  
}





export interface Variants {
    variant_id?: number;
    id?: number;
    sku?: string;
    size?: string;
    color?: string;
    image_id?: number;
}

export interface Images {
    image_id?: number;
    id?: number;
    alt?: string;
    src?: string;
    variant_id?: any[];
}